WordPress/src/main/java/org/wordpress/android/ui/posts/PostListMainViewModel.kt

```diff
@@ -35,4 +73,24 @@ class PostListMainViewModel @Inject constructor() : ViewModel() {
         val currentPage = POST_LIST_PAGES[position]
         _isFabVisible.value = FAB_VISIBLE_POST_LIST_PAGES.contains(currentPage)
     }
+
+    fun showTargetPost(targetPostId: Int) {
+        val postModel = postStore.getPostByLocalPostId(targetPostId)
+        if (postModel == null) {
+            _snackBarMessage.value = UiStringRes(string.error_post_does_not_exist)
+        } else {
+            launch(mainDispatcher) {
+                val targetTab = PostListType.fromPostStatus(PostStatus.fromPost(postModel))
+                _selectTab.value = POST_LIST_PAGES.indexOf(targetTab)
+                // we need to make sure the ViewPager initializes the targetTab fragment before we can propagate
+                // the targetPostId to it
+                withContext(bgDispatcher) {
+                    delay(SCROLL_TO_DELAY)
+                }
+                _scrollToLocalPostId.value = postModel.id
+            }
+        }
+    }
+
+    data class NavigationTarget(val localPostId: Int, val targetTabPosition: Int)
```

Comment:
This doesn't seem to be used anywhere. Did you maybe wanted to combine the tab selection and scroll and decided against it later?

Reply:
👍

Reply:
https://github.com/wordpress-mobile/WordPress-Android/pull/9301/commits/a79ba56e1f11d1fefc5db8df61dda85ff3df4b67